src/main/java/seedu/address/model/Model.java

```diff
@@ -258,4 +258,28 @@ public interface Model {
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredTaskList(Predicate<Task> predicate);
+
+    /** Returns true if contact list is being shown. Returns false otherwise. */
+    boolean isOnContactsScreen();
+
+    /** Returns true if events list is being shown. Returns false otherwise. */
+    boolean isOnEventsScreen();
+
+    /** Returns true if tags list is being shown. Returns false otherwise. */
+    boolean isOnTagsScreen();
+
+    /**
+     * Changes the boolean value of isOnContactsScreen to the given {@code isOnContactsScreen}.
+     */
+    void switchToContactsScreen(boolean onContactsScreen);
```

Comment:
is it necessary to have an input `boolean` parameter? Perhaps all switches can just be a standalone function that toggles screens only if its called rather than calling it everywhere and giving it a false boolean input if not required